wradlib/io.py

```diff
@@ -1616,53 +1610,51 @@ def get_RB_blobs_from_file(filename, rbdict):
 
     blobs = list(find_key('@blobid', rbdict))
 
-    datastring = get_RB_file_as_string(filename)
+    datastring = get_RB_file_as_string(fid)
     for blob in blobs:
         data = get_RB_blob_from_string(datastring, blob)
         blob['data'] = data
 
     return rbdict
 
 
-def get_RB_header(filename):
+def get_RB_header(fid):
     """Read Rainbow Header from filename, converts it to a dict and returns it
 
     Parameters
     ----------
-    filename : string
-        Filename of Data File
+    fid : file handle
+        File handle of Data File
 
     Returns
     -------
     object : dictionary
         Rainbow File Contents
 
     """
-    try:
-        fid = open(filename, "rb")
-    except IOError:
-        print("WRADLIB: Error opening Rainbow file ", filename)
-        raise IOError
 
     # load the header lines, i.e. the XML part
     endXMLmarker = b"<!-- END XML -->"
     header = b""
     line = b""
-    while not line.startswith(endXMLmarker):
-        header += line[:-1]
-        line = fid.readline()
-        if len(line) == 0:
-            break
 
-    fid.close()
+    try:
+        while not line.startswith(endXMLmarker):
+            header += line[:-1]
+            line = fid.readline()
+            if len(line) == 0:
+                break
+    except:
+        raise IOError('Could not read from file handle')
 
     xmltodict = util.import_optional('xmltodict')
 
     return xmltodict.parse(header)
 
 
-def read_Rainbow(filename, loaddata=True):
-    """"Reads Rainbow files files according to their structure
+@util.apichange_kwarg('0.9.2', par='filename', typ=str, expar='f')

```

Comment:
Removed the decorator according to your request. But isn't the decorator also mean to catch the case where people have `read_Rainbow(filename='my_file.azi')` in their code?